lib/batch_loader.rb

```diff
@@ -8,7 +8,7 @@
 require_relative "./batch_loader/graphql"
 
 class BatchLoader
-  IMPLEMENTED_INSTANCE_METHODS = %i[object_id __id__ __send__ singleton_method_added __sync respond_to? batch inspect].freeze
+  IMPLEMENTED_INSTANCE_METHODS = %i[object_id __id__ __send__ singleton_method_added __sync respond_to? batch inspect method_missing].freeze
```

Comment:
I had to add this for the spec, because if `method_missing` is undefined at the bottom of the class here, we can't stub it. I don't love that solution, though, so I'm open to a better spec!

I can't stub `__replace_with!` on the instance at all, even by adding it here, because trying to stub it loads the batch and calls `__replace_with!`. I could potentially stub it on `any_instance_of` but I don't like that 😕

Reply:
yeah, good point. 

will this work without changing `IMPLEMENTED_INSTANCE_METHODS`?

```rb
user_lazy = post.user_lazy(cache: true, replace_methods: false)

expect(user_lazy).to receive(:method_missing)

user_lazy.id
```

Reply:
You're right, that's fine! I think I confused myself because I was originally trying to stub `__replace_with!` but this passes with this change, and fails without it.